<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>899</width>
    <height>670</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>vga_test</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="gControl">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>291</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>Control</string>
    </property>
    <widget class="QSpinBox" name="clkPeriod">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>45</y>
       <width>60</width>
       <height>23</height>
      </rect>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="minimum">
      <number>5</number>
     </property>
     <property name="maximum">
      <number>2000</number>
     </property>
     <property name="singleStep">
      <number>5</number>
     </property>
     <property name="value">
      <number>100</number>
     </property>
    </widget>
    <widget class="MLed" name="clk">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>29</y>
       <width>10</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="assets.qrc">:/assets/leds/assets/leds/green_off.svg</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>29</y>
       <width>30</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string>clk</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="pbB">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>25</y>
       <width>60</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Run (F5)</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pbA">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>45</y>
       <width>60</width>
       <height>22</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Advance a clock tick</string>
     </property>
     <property name="text">
      <string>Tick (F3)</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pbReset">
     <property name="geometry">
      <rect>
       <x>223</x>
       <y>45</y>
       <width>60</width>
       <height>22</height>
      </rect>
     </property>
     <property name="text">
      <string>Reset</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="MLed" name="led_i_rst">
     <property name="geometry">
      <rect>
       <x>225</x>
       <y>30</y>
       <width>10</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="assets.qrc">:/assets/leds/assets/leds/green_off.svg</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>236</x>
       <y>29</y>
       <width>30</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string>i_rst</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_28">
     <property name="geometry">
      <rect>
       <x>9</x>
       <y>70</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Monospace</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Time:</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="main_time">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>70</y>
       <width>61</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Monospace</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="text">
      <string>AA</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>237</x>
       <y>74</y>
       <width>30</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string>halt</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="MLed" name="led_halt">
     <property name="geometry">
      <rect>
       <x>226</x>
       <y>75</y>
       <width>10</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="assets.qrc">:/assets/leds/assets/leds/green_off.svg</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pbINSTR">
     <property name="geometry">
      <rect>
       <x>149</x>
       <y>45</y>
       <width>60</width>
       <height>22</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>Advance to next Instruction</string>
     </property>
     <property name="text">
      <string>Instr (F4)</string>
     </property>
    </widget>
   </widget>
   <widget class="QPushButton" name="pbSave">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>20</y>
      <width>60</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Save</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pbLoad">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>50</y>
      <width>60</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>Load</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="g_Leds">
    <property name="geometry">
     <rect>
      <x>810</x>
      <y>0</y>
      <width>81</width>
      <height>51</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="title">
     <string>Leds</string>
    </property>
    <widget class="QLabel" name="R_LEDs">
     <property name="geometry">
      <rect>
       <x>59</x>
       <y>34</y>
       <width>20</width>
       <height>12</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Monospace</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="text">
      <string>AA</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="MLed" name="led7">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>20</y>
       <width>10</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="assets.qrc">:/assets/leds/assets/leds/green_off.svg</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="MLed" name="led6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>10</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="assets.qrc">:/assets/leds/assets/leds/green_off.svg</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="MLed" name="led5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>10</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="assets.qrc">:/assets/leds/assets/leds/green_off.svg</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="MLed" name="led4">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>20</y>
       <width>10</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="assets.qrc">:/assets/leds/assets/leds/green_off.svg</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="MLed" name="led3">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>20</y>
       <width>10</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="assets.qrc">:/assets/leds/assets/leds/green_off.svg</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="MLed" name="led2">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>20</y>
       <width>10</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="assets.qrc">:/assets/leds/assets/leds/green_off.svg</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="MLed" name="led1">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>20</y>
       <width>10</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="assets.qrc">:/assets/leds/assets/leds/green_off.svg</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="MLed" name="led0">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>20</y>
       <width>10</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="assets.qrc">:/assets/leds/assets/leds/green_off.svg</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="screen">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>110</y>
      <width>642</width>
      <height>482</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="text">
     <string>screen</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QGroupBox" name="gVGA">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>-1</y>
      <width>120</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>Vga</string>
    </property>
    <widget class="MLed" name="led_activevideo">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>10</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="pixmap">
      <pixmap resource="assets.qrc">:/assets/leds/assets/leds/green_off.svg</pixmap>
     </property>
     <property name="scaledContents">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>21</x>
       <y>29</y>
       <width>30</width>
       <height>10</height>
      </rect>
     </property>
     <property name="text">
      <string>vga</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="px_x">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Monospace</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="text">
      <string>AA</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="px_y">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>50</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Monospace</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="text">
      <string>AA</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="vc">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>70</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Monospace</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="text">
      <string>AA</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="hc">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Monospace</family>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="text">
      <string>AA</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>899</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuMenu">
    <property name="title">
     <string>&amp;Menu</string>
    </property>
    <addaction name="actionLoad_Program"/>
    <addaction name="actionGenerate_Trace"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuMenu"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionExit"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionLoad_Program">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>&amp;Load Program</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
  <action name="actionGenerate_Trace">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Generate Trace</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MLed</class>
   <extends>QLabel</extends>
   <header location="global">mled.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="assets.qrc"/>
 </resources>
 <connections/>
</ui>
